app/uploaders/avatar_uploader.rb

```diff
@@ -29,6 +29,11 @@ def extension_whitelist
   end
 
   def filename
-    "#{model.full_name.parameterize}-avatar" if original_filename
+    @name ||= "#{model.full_name.parameterize}-avatar-#{timestamp}" if original_filename
+  end
+
+  def timestamp
+    var = :"@#{mounted_as}_timestamp"
+    model.instance_variable_get(var) or model.instance_variable_set(var, Time.now.to_i)
```

Comment:
Just reusing the sample code provided by Carrierwave here, in case you're wondering :D